app/models/attachment.rb

```diff
@@ -13,6 +13,49 @@ class Attachment < ActiveRecord::Base
   #   The path to the attachment contents.
   delegate :url, :path, to: :file_upload
 
+  class << self
+    # Calculates the SHA256 hash and return the existing attachment which has the same hash.
+    # Otherwise a new attachment will be built.
+    #
+    # @param [Hash] attributes The hash attributes with the file.
+    # @return [Attachment] The attachment which contains the file.
+    def find_or_initialize_by(attributes, &block)
+      file = attributes[:file]
+      if file
+        digest = file_digest(file)
```

Comment:
I can't find a use case for that( find_or_create_by also cannot utilise it because `create` still needs the hash to be calculated). maybe I add that when there is a need ?